src/qml/qmlapplication.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <coreservices.h>
+
+#include <QApplication>
+#include <QFileSystemWatcher>
+#include <QQmlApplicationEngine>
+
+namespace mixxx {
+namespace qml {
+
+class QmlApplication : public QObject {
+    Q_OBJECT
+  public:
+    QmlApplication(
+            QApplication* app,
+            std::shared_ptr<CoreServices> pCoreServices);
+    ~QmlApplication() = default;
+
+  public slots:
+    void loadQml(const QString& path);
+
+  private:
+    std::shared_ptr<CoreServices> m_pCoreServices;
+
+    QString m_mainQmlFilePath;
+
+    std::unique_ptr<QQmlApplicationEngine> m_pQmlAppEngine;
+    QFileSystemWatcher m_qmlFileWatcher;
```

Comment:
Shall we drop the redudndant "qml/""Qml" term from all members? It seems to be redundant. This class should not take more responsibilities so it might to be safe.

Reply:
Yes, good idea.

Reply:
Should remove the `Qml` prefix from the class names within the `mixxx::qml` namespace?

Reply:
We could decide about the naming later. Let's leave it as is until we are sure.